src/common/core/core.js

```diff
@@ -1659,6 +1663,173 @@ define([
+         * version, keeping as much information as possible regarding the instances.
+         * @param {module:Core~Node} node - any regular node in your project.
+         * @param {string} name - the name of the library you want to update.
+         * @param {string} libraryRootHash - the hash of your library's new root
+         * (must exist in the project's collection at the time of call).
+         * @param {object} libraryInfo - information about your project.
+         * @param {string} libraryInfo.projectId - the projectId of your library.
+         * @param {string} libraryInfo.branchName - the branch that your library follows in the origin project.
+         * @param {string} libraryInfo.commitHash - the version of your library.
+         * @param {function()} callback
+         *
+         * @func
+         */
+        this.updateLibrary = core.updateLibrary;
+
+        /**
+         * Gives back the list of libraries in your project.
+         *
+         * @param {module:Core~Node} node - any node in your project.
+         *
+         * @return {string[]} - Returns the fully qualified names of all the libraries in your project
+         * (even embedded ones).
+         *
+         * @func
+         */
+        this.getLibraryNames = core.getLibraryNames;
+
+        /**
+         * Return the root of the inheritance chain of your Meta nodes.
+         *
+         * @param {module:Core~Node} node - any node in your project.
+         *
+         * @return {module:Core~Node} - Returns the acting FCO of your project.
+         *
+         * @func
+         */
+        this.getFCO = core.getFCO;
+
+        /**
+         * Returns if the node in question is a library root or not.
+         *
+         * @param {module:Core~Node} node - the node in question.
+         *
+         * @return {bool} - Returns true if your node is a library root (even if it is embedded in other library),
+         * false otherwise.
+         *
+         * @func
+         */
+        this.isLibraryRoot = core.isLibraryRoot;
+
+        /**
+         * Returns if the node in question is a library element or not.
+         *
+         * @param {module:Core~Node} node - the node in question.
+         *
+         * @return {bool} - Returns true if your node is a library element, false otherwise.
+         *
+         * @func
+         */
+        this.isLibraryElement = core.isLibraryElement;
+
+        /**
+         * Returns the fully qualified name of any node, which is the list of its namespaces separated
+         * by comma and followed by the name of the node.
+         *
+         * @param {module:Core~Node} node - the node in question.
+         *
+         * @return {string} - Returns the fully qualified name of the node (like: 'library.myName').
+         *
+         * @func
+         */
+        this.getFullyQualifiedName = core.getFullyQualifiedName;
+
+        /**
```

Comment:
we can iterate on the names, for now I made them consistent in the sense that they always get an arbitrary node as an input, and then a library name